serf/serf.go

```diff
@@ -25,7 +25,7 @@ import (
 // version to memberlist below.
 const (
 	ProtocolVersionMin uint8 = 2
-	ProtocolVersionMax       = 4
+	ProtocolVersionMax       = 5
```

Comment:
We will need to double check and see if this affects Consul and makes it harder to upgrade. For Consul we are "speaking" 2 but we "understand" 3 or whatever it is. If this complicates the upgrade we may want to do similar here, since we are switching based on what the receiver understands, so it's safe to run in a mix.